validator/accounts/account_test.go

```diff
@@ -155,3 +176,267 @@ func TestChangePassword_KeyNotMatchingOldPasswordNotEncryptedWithNewPassword(t *
+	if err := secondStore.SaveProposalHistoryForEpoch(context.Background(), secondStorePubKeys[1][:], proposalEpoch, proposalHistory4); err != nil {
+		return nil, errors.Wrapf(err, "Saving proposal history failed")
+	}
+
+	attestationHistoryMap1 := make(map[uint64]uint64)
+	attestationHistoryMap1[0] = 0
+	pubKeyAttestationHistory1 := &slashpb.AttestationHistory{
+		TargetToSource:     attestationHistoryMap1,
+		LatestEpochWritten: 0,
+	}
+	attestationHistoryMap2 := make(map[uint64]uint64)
+	attestationHistoryMap2[0] = 1
+	pubKeyAttestationHistory2 := &slashpb.AttestationHistory{
+		TargetToSource:     attestationHistoryMap2,
+		LatestEpochWritten: 0,
+	}
+	dbAttestationHistory1 := make(map[[48]byte]*slashpb.AttestationHistory)
+	dbAttestationHistory1[firstStorePubKeys[0]] = pubKeyAttestationHistory1
+	dbAttestationHistory1[firstStorePubKeys[1]] = pubKeyAttestationHistory2
+	if err := firstStore.SaveAttestationHistoryForPubKeys(context.Background(), dbAttestationHistory1); err != nil {
+		return nil, errors.Wrapf(err, "Saving attestation history failed")
+	}
+	attestationHistoryMap3 := make(map[uint64]uint64)
+	attestationHistoryMap3[0] = 2
+	pubKeyAttestationHistory3 := &slashpb.AttestationHistory{
+		TargetToSource:     attestationHistoryMap3,
+		LatestEpochWritten: 0,
+	}
+	attestationHistoryMap4 := make(map[uint64]uint64)
+	attestationHistoryMap4[0] = 3
+	pubKeyAttestationHistory4 := &slashpb.AttestationHistory{
+		TargetToSource:     attestationHistoryMap4,
+		LatestEpochWritten: 0,
+	}
+	dbAttestationHistory2 := make(map[[48]byte]*slashpb.AttestationHistory)
+	dbAttestationHistory2[secondStorePubKeys[0]] = pubKeyAttestationHistory3
+	dbAttestationHistory2[secondStorePubKeys[1]] = pubKeyAttestationHistory4
+	if err := secondStore.SaveAttestationHistoryForPubKeys(context.Background(), dbAttestationHistory2); err != nil {
+		return nil, errors.Wrapf(err, "Saving attestation history failed")
+	}
+
+	mergeHistory := &sourceStoresHistory{
+		ProposalEpoch:                       proposalEpoch,
+		FirstStoreFirstPubKeyProposals:      proposalHistory1,
+		FirstStoreSecondPubKeyProposals:     proposalHistory2,
+		SecondStoreFirstPubKeyProposals:     proposalHistory3,
+		SecondStoreSecondPubKeyProposals:    proposalHistory4,
+		FirstStoreFirstPubKeyAttestations:   attestationHistoryMap1,
+		FirstStoreSecondPubKeyAttestations:  attestationHistoryMap2,
+		SecondStoreFirstPubKeyAttestations:  attestationHistoryMap3,
+		SecondStoreSecondPubKeyAttestations: attestationHistoryMap4,
+	}
+
+	return mergeHistory, nil
+}
+
+func assertMergedStore(
+	t *testing.T,
+	mergedStore *db.Store,
+	firstStorePubKeys [][48]byte,
+	secondStorePubKeys [][48]byte,
+	history *sourceStoresHistory) {
+
+	mergedProposalHistory1, err := mergedStore.ProposalHistoryForEpoch(
+		context.Background(), firstStorePubKeys[0][:], history.ProposalEpoch)
+	if err != nil {
+		t.Errorf("Retrieving merged proposal history failed for public key %v", firstStorePubKeys[0])
+	} else {
```

Comment:
If I understand correctly how `t.Errorf` works, the function will continue execution. Without the `else` clause the test will try to compare the actual proposal history with the expected one. But since there was an error while retrieving the actual history in the first place, this check will always result in an error.

Reply:
If I understand correctly how `Errorf` works, the function will continue executing after the `Errorf` call. Without the `else` clause the test will try to compare the actual proposal history with the expected one. But since there was an error while retrieving the actual history in the first place, the comparison will always fail.

Reply:
from the function naming I would assume that it would be a fatal call. As 
it is `assertMergedStore` . If you do plan to only use `Errorf` do change the naming then.